src/core/qgsrulebasedlabeling.h

```diff
@@ -325,7 +325,7 @@ class CORE_EXPORT QgsRuleBasedLabeling : public QgsAbstractVectorLayerLabeling
 
       private:
         Rule *mParent; // parent rule (NULL only for root rule)
-        QgsPalLayerSettings *mSettings = nullptr;
+        QScopedPointer<QgsPalLayerSettings> mSettings;
```

Comment:
Use std::unique_ptr instead of the unmaintained qt smart pointers

Reply:
+1

Reply:
Oh yes ! I forgot we can use unique_ptr now. Good ! and thanks :)